src/main/webapp/app/shared/firebase/input/RealtimeBasicInput.tsx

```diff
@@ -0,0 +1,108 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Input, Label } from 'reactstrap';
+import { InputType } from 'reactstrap/es/Input';
+import classnames from 'classnames';
+import { ExtractPathExpressions } from '../../util/firebase/firebase-crud-store';
+import { Gene } from '../../model/firebase/firebase.model';
+import { IRootStore } from 'app/stores';
+import { RealtimeInputType } from './FirebaseRealtimeInput';
+import { inject } from 'mobx-react';
+import { getFirebasePath, getValueByNestedKey } from 'app/shared/util/firebase/firebase-utils';
+
+export type RealtimeBasicInput = Exclude<RealtimeInputType, RealtimeInputType.DROPDOWN>;
+
+export interface IRealtimeBasicInput extends React.InputHTMLAttributes<HTMLInputElement>, StoreProps {
+  fieldKey: ExtractPathExpressions<Gene>;
+  type: RealtimeBasicInput;
+  label: string;
+  labelClass?: string;
+  inputClass?: string;
+}
+
+const RealtimeBasicInput: React.FunctionComponent<IRealtimeBasicInput> = (props: IRealtimeBasicInput) => {
+  const {
+    fieldKey,
+    type,
+    onChange,
+    className,
+    labelClass,
+    label,
+    id = fieldKey,
+    inputClass,
+    children,
+    data,
+    updateReviewableContent,
+    ...otherProps
+  } = props;
+
+  const isCheckType = type === RealtimeInputType.CHECKBOX || type === RealtimeInputType.RADIO;
+  const isInlineInputText = type === RealtimeInputType.INLINE_TEXT;
+
+  const labelComponent = label && (
```

Comment:
Let's save the component as RealtimeInputLabel so it can be shared with RealtimeDropdownInput.